src/main/java/seedu/address/commons/core/index/Index.java

```diff
@@ -4,13 +4,13 @@
 
 /**
  * Represents a zero-based or one-based index.
- *
+ * <p>
  * {@code Index} should be used right from the start (when parsing in a new user input), so that if the current
  * component wants to communicate with another component, it can send an {@code Index} to avoid having to know what
  * base the other component is using for its index. However, after receiving the {@code Index}, that component can
  * convert it back to an int if the index will not be passed to a different component again.
  */
-public class Index {
+public class Index implements Comparable<Index> {
     private int zeroBasedIndex;
```

Comment:
nit: functionally identical, but easier to read
```suggestion
    private static int zeroBasedIndex;
```
optional (feel free to ignore)

[EDIT: as mentioned to me, I blundered; I meant to suggest `private final int zeroBasedIndex` rather than `static`]